xscen/extract.py

```diff
@@ -315,6 +321,14 @@ def extract_dataset(
                         catalog[key].df["xrfreq"].iloc[0]
                         == variables_and_freqs[var_name]
                     ):
+                        if ensure_correct_time:
+                            counts = da.time.resample(time=xrfreq).count()
+                            if any(counts > 1):
+                                raise ValueError(
+                                    "Dataset is labelled as having a sampling frequency of "
+                                    f"{xrfreq}, but some periods have more than one data point."
+                                )
```

Comment:
```suggestion
                            if any(counts > 1):
                                raise ValueError(
                                    "Dataset is labelled as having a sampling frequency of "
                                    f"{xrfreq}, but some periods have more than one data point."
                                )
                            if any(counts.isnull()):
                                warnings.warn('The resampling count contains nans. There might be some missing data.')
```
If there is a whole in the data, this might be a way to catch it. Related to issue #70. I am working on a fix for the check in `search`, but I think we also need to implement a check in `extract`.

If we add this, we also need to import warnings at the top. I put a warning not an error because you might want to live with the hole. I'm not sure if this should be in the logger instead?

Reply:
Just realised that if there is a whole, this will fail on `da["time"] = counts.time`, because the shape won't be that same. Maybe, we need to raise an error instead of a warning.
```suggestion
                            if any(counts > 1):
                                raise ValueError(
                                    "Dataset is labelled as having a sampling frequency of "
                                    f"{xrfreq}, but some periods have more than one data point."
                                )
                            if any(counts.isnull()):
                                raise ValueError('The resampling count contains nans. There might be some missing data.')
```

Reply:
Indeed! We need this check. Otherwise we would haul all whole holes!

Reply:
ishhh j'ai de la misère avec mes homophones!

Reply:
ses pas grave